package.json

```
@@ -130,6 +130,14 @@
     "webpack": "^5.44.0",
     "webpack-dev-server": "^3.11.2"
   },
+  "peerDependencies": {
+    "typescript": ">=4.2.0"
```

This range will target anything above 4.2.0. Perhaps let's provide an upper version boundary?

```suggestion
    "typescript": ">=4.2.0 < 5"
```

It may be running ahead of ourselves but we shouldn't promise compatibility with _any_ version higher than 4.2.0.

---

And that is a **horrible suggestion** on my side. 

I've recently learned that [TypeScript doesn't follow semver](https://github.com/microsoft/TypeScript/issues/14116). What should be the major version is, actually, a "market" version. The next one, `.2.` is indeed the _major_ version. In other words, there are breaking changes between 4.2 and 4.3 and that's intended on the TypeScript's side. 

This means we should lock this peer dependency range to:
```suggestion
    "typescript": "4.2.x"
```